### PR TITLE
Support delete dataset version

### PR DESCRIPTION
**Pull request by SophieGuo410**

### CODE REVIEW

**Review by mikeyang923 — APPROVED** [review]
lgtm

File: ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java [hunk 1]
```diff
@@ -13,6 +13,10 @@
  */
 package com.github.ambry.frontend;
 
+import com.github.ambry.account.AccountService;
+import com.github.ambry.account.AccountServiceException;
+import com.github.ambry.account.Dataset;
+import com.github.ambry.account.DatasetVersionRecord;
```
Comment (justinlin-linkedin) [inline ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java:19]: not used anywhere.

File: ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java [hunk 2]
```diff
@@ -148,6 +156,15 @@ private Callback<Void> securityPostProcessRequestCallback(BlobId blobId) {
      */
     private Callback<Void> routerCallback() {
       return buildCallback(metrics.deleteBlobRouterMetrics, result -> {
+        if (RestUtils.isDatasetVersionUpload(restRequest.getArgs())) {
```
Comment (justinlin-linkedin) [inline ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java:159]: now that this header is used by many http methods, can we rename this header to something else?

**Review by justinlin-linkedin — APPROVED** [review]

File: ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java (hunk 1 above)
Comment (SophieGuo410) [reply ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java]: Removed.

File: ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java (hunk 2 above)
Comment (SophieGuo410) [reply ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java]: Rename to x-ambry-dataset-version-query-enabled.

**Pull request merged**